eva/optimizer/rules/rules.py

```diff
@@ -563,15 +584,18 @@ def __new__(cls):
         return cls._instance
 
     def __init__(self):
-        self._rewrite_rules = [
-            EmbedFilterIntoGet(),
+        self._logical_rules = [
             EmbedProjectIntoGet(),
-            EmbedFilterIntoDerivedGet(),
             EmbedProjectIntoDerivedGet(),
-            PushdownFilterThroughSample(),
             PushdownProjectThroughSample()
         ]
 
+        self._rewrite_rules = [
+            EmbedFilterIntoGet(),
+            EmbedFilterIntoDerivedGet(),
+            PushdownFilterThroughSample()
+        ]
+
```

Comment:
What is the difference between logical rules and rewrite rules?

Reply:
I'll fix the naming.